### PR TITLE
merge: develop -> master

### PR DESCRIPTION
**Pull request by lukewhrit**
This PR contains lots of various improvements over the server. Changes are detailed below:

## Changes

* Adds a usable Dockerfile (- @jackmerrill)
* Adds an OpenAPI specification (- @GreatGodApollo)
* Various cleanup
* Make API responses consistent
* Stop synchronizing production databases in TypeORM
* Refactored the Document Handler

### CODE REVIEW

**Review by jackiedorland — APPROVED** [review]
:+1:

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
-CMD [ "yarn", "start" ]
+FROM node:latest
+
+RUN git clone https://github.com/spacebin-org/server.git spacebin 
+
+WORKDIR spacebin
+
+RUN yarn add sqlite3
+
+EXPOSE 7777:7777
```
Comment (lukewhrit) [inline Dockerfile:9]: Codacy found an issue: [Dockerfile:9:12:
  |
9 | EXPOSE 7777:7777
  |            ^
unexpected ':'
expecting a new line followed by the next instruction, at least one space, or digit
](https://app.codacy.com/gh/spacebin-org/server/pullRequest?prid=5760887)

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
-CMD [ "yarn", "start" ]
+FROM node:latest
+
+RUN git clone https://github.com/spacebin-org/server.git spacebin
+
+WORKDIR spacebin
+
+RUN yarn add sqlite3
+
+EXPOSE 7777:7777
```
Comment (lukewhrit) [inline Dockerfile:9]: Codacy found an issue: [Dockerfile:9:12:
  |
9 | EXPOSE 7777:7777
  |            ^
unexpected ':'
expecting a new line followed by the next instruction, at least one space, or digit
](https://app.codacy.com/gh/spacebin-org/server/pullRequest?prid=5760887)

Comment (lukewhrit) [reply Dockerfile:9]: @jackmerrill

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,11 @@
-FROM node:14
-WORKDIR /opt/spacebin-api
-COPY package.json yarn.lock ./
-COPY . .
-RUN yarn
+FROM node:latest
+
+RUN git clone https://github.com/spacebin-org/server.git spacebin
+
+WORKDIR spacebin
```
Comment (lukewhrit) [inline Dockerfile:5]: Codacy found an issue: [Use absolute WORKDIR](https://app.codacy.com/gh/spacebin-org/server/pullRequest?prid=5760887)

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,11 @@
-FROM node:14
-WORKDIR /opt/spacebin-api
-COPY package.json yarn.lock ./
-COPY . .
-RUN yarn
+FROM node:latest
```
Comment (lukewhrit) [inline Dockerfile:1]: Codacy found an issue: [Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag](https://app.codacy.com/gh/spacebin-org/server/pullRequest?prid=5760887)

**Pull request merged**